diff --git a/iMage/jmjrst.main/src/main/java/org/jis/generator/Generator.java b/iMage/jmjrst.main/src/main/java/org/jis/generator/Generator.java
index 4309b7b..48399b1 100755
--- a/iMage/jmjrst.main/src/main/java/org/jis/generator/Generator.java
+++ b/iMage/jmjrst.main/src/main/java/org/jis/generator/Generator.java
@@ -743,6 +743,20 @@
       width = image.getHeight(null); // swap
       height = image.getWidth(null);
     }
+    else if (rotate == Generator.ROTATE_M90)
+    {
+      transform.translate(0, width);
+      transform.rotate(Generator.ROTATE_M90);
+      width = image.getHeight(); // swap
+      height = image.getWidth();
+    }
+    else if (rotate == Generator.ROTATE_M270)
+    {
+      transform.translate(height, 0);
+      transform.rotate(Generator.ROTATE_M270);
+      width = image.getHeight();
+      height = image.getWidth();
+    }
     else
     {
       throw new IllegalArgumentException("degree must be a mutiple of 90°!");
